### PR TITLE
ci: test on pypy

### PR DESCRIPTION
**Pull request by skshetry**

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 locations = "src", "tests"
 
 
-@nox.session(python=["3.7", "3.8", "3.9", "3.10"])
+@nox.session(python=["3.7", "3.8", "3.9", "3.10", "pypy-3.8", "pypy-3.9"])
```
Comment (skshetry) [inline noxfile.py:12]: Let's drop 3.7?

Comment (pared) [reply noxfile.py:12]: Yeah, are we doing it in this change or should we do follow up?

**Review by pared — APPROVED** [review]

File: noxfile.py (hunk 1 above)
Comment (skshetry) [inline noxfile.py:12]:
```suggestion
@nox.session(python=["3.7", "3.8", "3.9", "3.10", "pypy3.8", "pypy3.9"])
```

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -21,6 +21,9 @@ jobs:
       matrix:
         os: [ubuntu-20.04, windows-latest, macos-latest]
         pyv: ['3.7', '3.8', '3.9', '3.10']
+        include:
+        - os: ubuntu-latest
+          pyv: 'pypy-3.8'
```
Comment (skshetry) [inline .github/workflows/tests.yml:26]:
```suggestion
          pyv: 'pypy3.8'
```

File: noxfile.py (hunk 1 above)
Comment (skshetry) [reply noxfile.py]: Will do in a separate PR.

**Pull request merged**